### PR TITLE
Bring in E3SM cime changes for past 3 months

### PR DESCRIPTION
**Pull request by jgfouca**
Major change list:

**Python**
- Minor improvements to bless_test_results.py
- Add mach-specific throughput checking to test machines
- Some tools were not loading environment before running which was causing problems on some platforms
- Improvements in robustness, error msgs of check_input_data
- configure.py new FakeCase class
- add FakeCase to fortran_unit_testing/run_tests.py
- system_tests_common.py  change to tput test
- env_batch.py add new CHARGE_ACCOUNT
- env_mach_specific.py - Allow config_machines.xml to have any case variable as an attribute,
- pes.py - also output threads, root
- generic_xml.py - remove minidom
- build.py - add USE_MOAB option
- case.py - use new CHARGE_ACCOUNT.  new SMP_PRESENT
- provenancy.py - update E3SM provenance.
- test_scheduler.py - tput tolerance.  new "R" modifier for single column tests.
- utils.py new get_charge_account method
- update_acme_tests.py - more tests and test suites.

**schema changes**
- config_compilers_v2.xsd:  Add MOAB
- config_machines.xsd:  Add CHARGE_ACCOUNT for charging to a different name then PROJECT used for directory paths
- config_machines_template.xml:  Also add CHARGE_ACCOUNT
- env_mach_specific.xsd: Change debug to upperBoolean

**Tools**
- new acme_cime_split to help with subtree splitting
- new tests for the run_acme script in scripts/test/run_acme_test

**xml changes**
- Change "debug" to "DEBUG" in all config_machines.xml

** data models **
- datm/cime_config/namelist_definition_datm.xml new 6hourly forcing data set

** mct driver **
- config_component.xml  new SMP_PRESENT, USE_MOAB, TEST_TPUT_TOLERANCE, CHARGE_ACCOUNT
- cime_comp_mod.F90  new option to shr_mem_getusage to also print.
- seq_diag_mct.F90  fix an E3SM diag

**timing**
- fix for BGQ

shr_reprosum_mod.F90 - optionally check for nans.

Rename some man3 pages in PIO1

Many E3SM config changes.

Test suite: scripts_regression_tests
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes ESMCI/cime#1332
Fixes ESMCI/cime#1626

User interface changes?: 

Update gh-pages html  N

Code review:

### CODE REVIEW

**Comment by jgfouca** [comment]
Tests still running, please review in the meantime.

File: config/acme/machines/config_batch.xml [hunk 1]
```diff
@@ -35,27 +35,27 @@
...
+     <depend_string> --dependencies</depend_string>
+     <walltime_format>%H:%M:%s</walltime_format>
+     <batch_mail_flag>-M</batch_mail_flag>
+     <batch_mail_type_flag></batch_mail_type_flag>
+     <batch_mail_type></batch_mail_type>
+     <submit_args>
+       <arg flag="--cwd" name="CASEROOT"/>
+       <arg flag="-A" name="PROJECT"/>
+       <arg flag="-t" name="JOB_WALLCLOCK_TIME"/>
+       <arg flag="-n" name=" ($TOTALPES + $PES_PER_NODE - 1)/$PES_PER_NODE"/>
```
Comment (jgfouca) [inline config/acme/machines/config_batch.xml]: Did PES_PER_NODE get removed?

File: scripts/Testing/Testcases/ICP_script [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+touch cice_perf.out
+
+./xmlchange --file env_run.xml   --id CONTINUE_RUN      --val FALSE
+./xmlchange --file env_run.xml   --id REST_OPTION       --val none
+./xmlchange --file env_build.xml --id CICE_AUTO_DECOMP  --val false
+
+cat >! user_nl_cice.mhT <<EOF
+maskhalo_bound = .true.
+maskhalo_dyn = .true.
```
Comment (jgfouca) [inline scripts/Testing/Testcases/ICP_script]: I think Testing/Testcases/* can be removed.

File: scripts/Testing/testreporter.pl [hunk 3]
```diff
@@ -0,0 +1,678 @@
...
+use Data::Dumper;
+use LWP;
+use HTTP::Request;
+use HTTP::Request::Common qw(POST);
+use XML::LibXML;
+use Cwd qw(abs_path);
+#-------------------------------------------------------------------------------
+# testreporter.pl
+# Perl script that watches the CIME tests as they progress, and sends the reports to
+# the testdb application at csegweb.cgd.ucar.edu
```
Comment (jgfouca) [inline scripts/Testing/testreporter.pl]: Can this be removed?

File: scripts/lib/CIME/SystemTests/pet.py [hunk 4]
```diff
@@ -33,6 +33,10 @@ def _case_one_setup(self):
         # Need to redo case_setup because we may have changed the number of threads
         case_setup(self._case, reset=True)
 
+        # Hack until CIME queue selection is more dynamic
```
Comment (jgfouca) [inline scripts/lib/CIME/SystemTests/pet.py]: This can be removed.

File: scripts/lib/CIME/XML/env_mach_specific.py [hunk 5]
```diff
@@ -54,44 +54,43 @@ def populate(self, machobj):
...
+        self._get_resources_for_case(case)
 
-        self._get_resources_for_case(compiler, debug, mpilib)
-
-    def _get_resources_for_case(self, compiler, debug, mpilib):
+    def _get_resources_for_case(self, case):
         resource_nodes = self.get_nodes("resource_limits")
         if resource_nodes is not None:
-            nodes = self._compute_resource_actions(resource_nodes, compiler, debug, mpilib)
+            nodes = self._compute_resource_actions(resource_nodes, case)
```
Comment (jgfouca) [inline scripts/lib/CIME/XML/env_mach_specific.py]: Changes to this file are worth a close review.

File: scripts/lib/CIME/case_st_archive.py [hunk 6]
```diff
@@ -20,45 +20,143 @@ def _get_archive_file_fn(copy_only):
...
 ###############################################################################
-def _get_datenames(case, last_date=None):
+def _get_datenames(rundir, casename):
 ###############################################################################
     """
-    Returns a list of datenames giving the dates of cpl restart files
-
-    If there are no cpl restart files, this will return []
+    Returns the datetime objects specifying the times of each file
+    Note we are assuming that the coupler restart files exist and are consistent with other component datenames
```
Comment (jgfouca) [inline scripts/lib/CIME/case_st_archive.py]: @mfdeakin-sandia made a lot of changes to this file.

File: scripts/lib/CIME/test_scheduler.py [hunk 7]
```diff
@@ -526,8 +528,27 @@ def _xml_phase(self, test):
                     envtest.set_test_parameter("STOP_OPTION",stop_option[opt])
                     opti = match.group(2)
                     envtest.set_test_parameter("STOP_N", opti)
-                    logger.debug (" STOP_OPTION set to {}".format(stop_option[opt]))
-                    logger.debug (" STOP_N      set to {}".format(opti))
+
+                    logger.debug (" STOP_OPTION set to %s" %stop_option[opt])
```
Comment (jgfouca) [inline scripts/lib/CIME/test_scheduler.py]: These need to be converted back to python3-style.

File: src/drivers/mct/bld/testdir/env_build.xml [hunk 8]
```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0"?>
+<file id="env_build.xml">
+  <header>
+    These variables SHOULD NOT be changed once the model has been built.
+    urrently, these variables are not cached.
+    Note1: users SHOULD NOT modify BUILD_COMPETE below
+    this is done automatically by the scripts.
+    </header>
```
Comment (jgfouca) [inline src/drivers/mct/bld/testdir/env_build.xml]: Suspicious: seeing many new files.

File: src/externals/pio1/tests/testpio/perl5lib/XML/man3/XML_Lite.3 [hunk 9]
```diff
@@ -1,213 +0,0 @@
-.\" Automatically generated by Pod::Man version 1.02
-.\" Tue Mar 18 09:37:35 2003
-.\"
-.\" Standard preamble:
-.\" ======================================================================
-.de Sh \" Subsection heading
-.br
-.if t .Sp
```
Comment (jgfouca) [inline src/externals/pio1/tests/testpio/perl5lib/XML/man3/XML_Lite.3]: @jayeshkrishna should these have been removed?

File: src/externals/pio2/src/clib/ncparser.pl [hunk 10]
```diff
@@ -0,0 +1,362 @@
+#!/usr/bin/perl 
+use strict;
+
+my $netcdf_incdir = $ARGV[0];
+my $pnetcdf_incdir = $ARGV[1];
+
+my $netcdf_inc = "$netcdf_incdir/netcdf.h";
+my $pnetcdf_inc = "$pnetcdf_incdir/pnetcdf.h";
+
```
Comment (jgfouca) [inline src/externals/pio2/src/clib/ncparser.pl]: @jayeshkrishna

File: tests/unit/test_names.c [hunk 11]
```diff
@@ -0,0 +1,416 @@
...
+ */
+#include <pio.h>
+#ifdef TIMING
+#include <gptl.h>
+#endif
+
+#define NUM_NETCDF_FLAVORS 4
+#define NDIM 3
+#define X_DIM_LEN 400
+#define Y_DIM_LEN 400
```
Comment (jgfouca) [inline tests/unit/test_names.c]: What is this new file?

File: tools/mapping/gen_mapping_files/runoff_to_ocn/runoff_map_tx01.nml [hunk 12]
```diff
@@ -0,0 +1,15 @@
+  &input_nml
+   gridtype     = 'obs'
+   file_roff    = '/fs/cgd/csm/inputdata/lnd/clm2/rtmdata/rdirc.05.061026'
+   file_ocn     = '/fs/cgd/csm/mapping/grids/tx0.1v2_070911.nc'
+   file_nn      = 'map_nn_tx01v2.nc '
```
Comment (jgfouca) [inline tools/mapping/gen_mapping_files/runoff_to_ocn/runoff_map_tx01.nml]: More zombie files?

File: src/drivers/mct/bld/testdir/env_build.xml (hunk 8 above)
Comment (rljacob) [reply src/drivers/mct/bld/testdir/env_build.xml]: This entire directory "src/drivers/mct/bld" is not present in ESMCI cime.

File: src/externals/pio2/src/clib/ncparser.pl (hunk 10 above)
Comment (jedwards4b) [reply src/externals/pio2/src/clib/ncparser.pl]: why is this here?

**Comment by rljacob** [comment]
Regarding the src/drivers/mct/bld dir and its files, according to the ESMCI git history:
```
commit 193b7d0528a13fd70f2a880f3a2d3981e539b19e
Author: James Foucar <jgfouca@sandia.gov>
Date:   Fri Jun 9 12:42:17 2017 -0600

    Remove src/drivers/mct/bld
```

So how come when we subtree merged ESMCI to ACME, this directory was not deleted?

@jgfouca I think you should make a PR to ACME that removes this dir and redo the subtree split after its merged to master.

File: scripts/lib/CIME/XML/env_mach_specific.py (hunk 5 above)
Comment (jedwards4b) [reply scripts/lib/CIME/XML/env_mach_specific.py]: This is what allows config_machines.xml to have any case variable as an attribute, looks good.

File: src/drivers/mct/cime_config/config_component.xml [hunk 13]
```diff
@@ -1422,6 +1449,40 @@
     <desc>ocn2atm state mapping file decomp type</desc>
   </entry>
 
+  <entry id="OCN2GLC_FMAPNAME">
```
[the remark this answers is not in the document]
Comment (billsacks) [reply src/drivers/mct/cime_config/config_component.xml]: @jedwards4b You mean OCN to GLC? I have a feeling they're needed for ACME, even though they're not currently needed for CESM. Need to make sure this is done in a way that won't break CESM.

**Comment by billsacks** [comment]
Wow, this is huge. A few thoughts:

(1) @jgfouca Before this gets merged to master, can you please add documentation summarizing the changes that are coming in here? This documentation should make it into the merge commit.

(2) I'm really concerned about regressions, partly based on @jedwards4b 's findings. I'm not sure what to do about that: I don't think I have the energy to carefully review 8000+ lines of changes, though I certainly appreciate the time others have spent reviewing these changes so far! @jedwards4b @mvertens @fischer-ncar - I feel we should run the full CESM prebeta test suite on these changes (with baseline comparisons) as at least one additional sanity test.

(3) At today's cime call, I'd like to brainstorm how we can avoid big bang merges like this in the future.

**Comment by rljacob** [comment]
Regrading 3, this should be the last one of these giant merges. For various reasons, we were delayed in updating CIME on the E3SM side, which in turn delayed this update back.  We can increase the frequency after this which will make these much smaller.

File: scripts/lib/CIME/test_scheduler.py (hunk 7 above)
Comment (jedwards4b) [reply scripts/lib/CIME/test_scheduler.py]: Once you have completed testing with python2 and resolved the concerns raised in review I will run testing with python3.

**Comment by billsacks** [comment]
Okay, thanks @rljacob

**Comment by jgfouca** [comment]
@rljacob I was thinking the same think. ACME merges keep bringing deleted files back from the dead, so we can try deleting these files from ACME.

I'd rather not re-do the merge since I've already done significant conflict-resolving work. I'll just delete them in both places.

**Comment by jgfouca** [comment]
I apologize to the CSEG folks for dropping these massive merges. I know it's annoying for all parties involved. As Rob said, this should be the last one.

**Comment by billsacks** [comment]
Thanks, @jgfouca . And I realize that, however painful this is on our end, it must be much more so on your end....

File: config/config_tests.xml [hunk 14]
```diff
@@ -247,6 +247,7 @@ NODEFAIL          Tests restart upon detected node failure. Generates fake failu
     <STOP_OPTION>ndays</STOP_OPTION>
     <STOP_N>11</STOP_N>
     <DOUT_S>FALSE</DOUT_S>
+    <BUILD_THREADED>TRUE</BUILD_THREADED>
```
[the remark this answers is not in the document]
Comment (jgfouca) [reply config/config_tests.xml]: @jedwards4b isn't it better to do XML setting here than in the python?

File: config/xml_schemas/config_machines.xsd [hunk 15]
```diff
@@ -71,6 +72,9 @@
 	<!-- PROJECT: A project or account number used for batch jobs
 	  can be overridden in environment or $HOME/.cime/config -->
 	<xs:element ref="PROJECT" minOccurs="0" maxOccurs="1"/>
+	<!-- CHARGE_ACCOUNT: The name of the account to charge for batch jobs
+	  can be overridden in environment or $HOME/.cime/config -->
+	<xs:element ref="SUBPROJECT" minOccurs="0" maxOccurs="1"/>
```
[the remark this answers is not in the document]
Comment (jgfouca) [reply config/xml_schemas/config_machines.xsd]: @jedwards4b good catch. I think they meant to put CHARGE_PROJECT here instead of SUBPROJECT. I have no idea what a SUBPROJECT is.

File: config/xml_schemas/env_mach_specific.xsd [hunk 16]
```diff
@@ -5,7 +5,7 @@
 <xs:attribute name="version" type="xs:decimal"/>
 <xs:attribute name="lang" type="xs:NCName"/>
 <xs:attribute name="compiler" type="xs:string"/>
-<xs:attribute name="debug" type="xs:boolean"/>
+<xs:attribute name="DEBUG" type="upperBoolean"/>
```
Comment (jgfouca) [inline config/xml_schemas/env_mach_specific.xsd]: @jedwards4b I don't know why this change was made (changing debug to DEBUG). Is that going to break you guys?

Comment (jgfouca) [reply config/xml_schemas/env_mach_specific.xsd]: @amametjanov , looks like you made this change. Care to chime in?

File: scripts/Testing/Testcases/ICP_script [hunk 17]
```diff
@@ -0,0 +1,108 @@
+
```
[the remark this answers is not in the document]
Comment (jgfouca) [reply scripts/Testing/Testcases/ICP_script]: Nope. zombies

File: scripts/lib/CIME/SystemTests/erp.py [hunk 18]
```diff
@@ -27,9 +27,6 @@ def __init__(self, case):
                              run_one_description = 'initial',
                              run_two_description = 'restart')
 
-    def _common_setup(self):
```
[the remark this answers is not in the document]
Comment (jgfouca) [reply scripts/lib/CIME/SystemTests/erp.py]: wasn't needed once the XML did this for us.

File: scripts/lib/CIME/SystemTests/pet.py [hunk 19]
```diff
@@ -49,3 +53,7 @@ def _case_two_setup(self):
 
         # Need to redo case_setup because we may have changed the number of threads
         case_setup(self._case, reset=True)
+
+        # Hack until CIME queue selection is more dynamic
```
[the remark this answers is not in the document]
Comment (jgfouca) [reply scripts/lib/CIME/SystemTests/pet.py]: don't sweat it, it's gone.

File: config/xml_schemas/config_machines.xsd (hunk 15 above)
[the remark this answers is not in the document]
Comment (rljacob) [reply config/xml_schemas/config_machines.xsd]: This is our solution to https://github.com/ESMCI/cime/issues/1332 which was "fixed in ACME" and now being brought in here.

File: config/xml_schemas/env_mach_specific.xsd (hunk 16 above)
Comment (jedwards4b) [reply config/xml_schemas/env_mach_specific.xsd]: Isn't this tied to the changes in env_mach_specific?  It will require some changes in the config_machines.xml file on the cesm side, but I think that it is a good change.

File: scripts/lib/CIME/XML/generic_xml.py [hunk 20]
```diff
@@ -79,9 +79,8 @@ def write(self, outfile=None):
         if xmllint is not None:
             run_cmd_no_fail("{} --format --output {} -".format(xmllint, outfile), input_str=xmlstr)
         else:
-            doc = minidom.parseString(xmlstr)
             with open(outfile,'w') as xmlout:
-                doc.writexml(xmlout,addindent='  ')
+                xmlout.write(xmlstr)
```
[the remark this answers is not in the document]
Comment (jgfouca) [reply scripts/lib/CIME/XML/generic_xml.py]: @jedwards4b it's been a while, but I think I was having problems with minidom on some machines. I checked the new output and still looked pretty readable. We can revisit this if needed.

File: scripts/lib/CIME/case_st_archive.py (hunk 6 above)
Comment (jgfouca) [reply scripts/lib/CIME/case_st_archive.py]: @mfdeakin-sandia please help

File: scripts/lib/CIME/test_scheduler.py [hunk 21]
```diff
@@ -526,8 +528,27 @@ def _xml_phase(self, test):
...
+                    envtest.set_test_parameter("PTS_MODE", "TRUE")
+
+                    # For PTS_MODE, compile with mpi-serial
+                    envtest.set_test_parameter("MPILIB", "mpi-serial")
+                    comps=["ATM","LND","ICE","OCN","CPL","GLC","ROF","WAV"]
+                    for comp in comps:
+                        envtest.set_test_parameter("NTASKS_"+comp,"1")
+
+                    # Set latitude and longitude for the appropriate case
+                    # Below for ARM97, default SCM test case
```
[the remark this answers is not in the document]
Comment (jgfouca) [reply scripts/lib/CIME/test_scheduler.py]: @jedwards4b can you elaborate? I don't know how to do this.

File: scripts/lib/CIME/case_st_archive.py (hunk 6 above)
Comment (mfdeakin-sandia) [reply scripts/lib/CIME/case_st_archive.py]: I'll look into it - are there tests for the multi driver code, so I can make certain I put them back correctly?

File: scripts/lib/CIME/XML/generic_xml.py (hunk 20 above)
[the remark this answers is not in the document]
Comment (jedwards4b) [reply scripts/lib/CIME/XML/generic_xml.py]: maybe a try, except construct?

File: scripts/lib/CIME/case_st_archive.py (hunk 6 above)
Comment (jedwards4b) [reply scripts/lib/CIME/case_st_archive.py]: yes, MCC in cime_developer is one.

File: config/acme/machines/config_batch.xml (hunk 1 above)
Comment (jedwards4b) [reply config/acme/machines/config_batch.xml]: It was renamed MAX_MPITASKS_PER_NODE

File: config/config_tests.xml (hunk 14 above)
[the remark this answers is not in the document]
Comment (jedwards4b) [reply config/config_tests.xml]: ok

File: src/share/util/mct_mod.F90 [hunk 22]
```diff
@@ -98,6 +98,7 @@ module mct_mod
    use m_GlobalSegMap       ,only: mct_gsMap_gstorage     => GlobalStorage
    use m_GlobalSegMap       ,only: mct_gsMap_ngseg        => ngseg
    use m_GlobalSegMap       ,only: mct_gsMap_nlseg        => nlseg
+   use m_GlobalSegMap       ,only: mct_gsMap_OP           => OrderedPoints
```
Comment (jedwards4b) [inline src/share/util/mct_mod.F90]: This line was removed because it was a redefinition of mct_gsMap_orderedPoints on line 106.  I believe you need to make a change in the land model from mct_geMap_OP to mct_gsMap_orderedPoints

File: tools/mapping/gen_mapping_files/runoff_to_ocn/runoff_map_tx01.nml (hunk 12 above)
Comment (jedwards4b) [reply tools/mapping/gen_mapping_files/runoff_to_ocn/runoff_map_tx01.nml]: I think so

**Comment by jgfouca** [comment]
TODO:

1. Add description to PR
2. case_st_archive
3. @jayeshkrishna to look at PIO
4. CSEG run a pre-beta test
5. Bring-in st_archiver changes as separate PR?
6. mct_mod change
7. CESM config_machines.xml needs to be changed to DEBUG

File: src/externals/pio2/src/clib/ncputgetparser.pl [hunk 23]
```diff
@@ -0,0 +1,352 @@
+#!/usr/bin/perl
```
Comment (jayeshkrishna) [inline src/externals/pio2/src/clib/ncputgetparser.pl]: Same here, ncputgetparser.pl was also deleted (Sep 17, 2016) from pio2 master (https://github.com/PARALLELIO/ParallelIO.git) and is no longer needed

File: src/externals/pio2/src/clib/pio_c_get_template.c [hunk 24]
```diff
@@ -0,0 +1,62 @@
+int PIO_function()
```
Comment (jayeshkrishna) [inline src/externals/pio2/src/clib/pio_c_get_template.c]: Same here, pio_c_get_template.c was also deleted (Sep 17, 2016) from pio2 master (https://github.com/PARALLELIO/ParallelIO.git) and is no longer needed

File: src/externals/pio2/src/clib/pio_c_put_template.c [hunk 25]
```diff
@@ -0,0 +1,77 @@
+///
```
Comment (jayeshkrishna) [inline src/externals/pio2/src/clib/pio_c_put_template.c]: Same here, pio_c_put_template.c was also deleted (Sep 17, 2016) from pio2 master (https://github.com/PARALLELIO/ParallelIO.git) and is no longer needed

File: src/externals/pio2/src/clib/pio_c_template.c [hunk 26]
```diff
@@ -0,0 +1,54 @@
+int PIO_function()
```
Comment (jayeshkrishna) [inline src/externals/pio2/src/clib/pio_c_template.c]: Same here, pio_c_template.c was also deleted (Sep 17, 2016) from pio2 master (https://github.com/PARALLELIO/ParallelIO.git) and is no longer needed

**Comment by jgfouca** [comment]
@jedwards4b @mvertens I think this is ready to go. Please do your prebeta testing and whatever other testing you think you need to do.

File: scripts/lib/CIME/test_scheduler.py [hunk 27]
```diff
@@ -526,8 +528,24 @@ def _xml_phase(self, test):
                     envtest.set_test_parameter("STOP_OPTION",stop_option[opt])
                     opti = match.group(2)
                     envtest.set_test_parameter("STOP_N", opti)
+
                     logger.debug (" STOP_OPTION set to {}".format(stop_option[opt]))
                     logger.debug (" STOP_N      set to {}".format(opti))
+
+                elif opt.startswith('R'):
```
Comment (rljacob) [inline scripts/lib/CIME/test_scheduler.py]: @jedwards4b and think you said we don't need this "R" option but its doing more then setting mpi-serial.

**Comment by rljacob** [comment]
I did a major update to the description.

File: config/cesm/machines/config_machines.xml [hunk 28]
```diff
@@ -307,16 +307,16 @@
 	<command name="load">esmf_libs</command>
 	<command name="load">mkl</command>
       </modules>
-      <modules compiler="intel" mpilib="!mpi-serial" debug="true">
+      <modules compiler="intel" mpilib="!mpi-serial" DEBUG="true">
```
Comment (jedwards4b) [inline config/cesm/machines/config_machines.xml]: I believe that the TRUE FALSE also needs to be upcased.

File: scripts/lib/CIME/test_scheduler.py (hunk 27 above)
Comment (jedwards4b) [reply scripts/lib/CIME/test_scheduler.py]: I think that the R is fine, but the hardcoded A97 case at lines 545-547 are not.   On the cesm side these have been moved to components/cam/cime_config

Comment (rljacob) [reply scripts/lib/CIME/test_scheduler.py]: Which file in cam/cime_config does this go?

Comment (jedwards4b) [reply scripts/lib/CIME/test_scheduler.py]: config_compsets.xml

**Comment by jedwards4b** [comment]
There was a problem in the schema file for env_mach_specific.xml - the attribute modifiers were defined for both environment_variables and for env - you can't have both (okay technically you can but it makes for a logic nightmare) I have removed the attributes from the env field -  this will probably require changes in the acme config_machines.xml file.

**Comment by jgfouca** [comment]
@jedwards4b , we've already removed all the attribute modifiers for individual environs in ACME.

**Comment by jedwards4b** [comment]
Good, no problem then.

**Comment by jgfouca** [comment]
@jedwards4b I've addressed your feedback regarding the 'R' options.

@rljacob I've left a comment in test_scheduler that will hopefully remind us to make the needed changes once this gets back to ACME.

**Comment by rljacob** [comment]
ok

**Comment by jgfouca** [comment]
@jedwards4b OK, i will go ahead and merge then.
